ros2sim/parsers/__init__.py

```diff
@@ -1,12 +1,152 @@
-  ACTION = b'ACTION'
-  RESET = b'RESET'
-  OBS_REQUEST = b'OBS'
-  OK = b'OK'
 
+class SerialReadState(Enum):
+  INIT = auto()
+  READ_PREAMBLE = auto()
+  READ_DATA_REQUEST = auto()
+  READ_L_HIP = auto()
+  READ_L_KNEE = auto()
+  READ_R_HIP = auto()
+  READ_R_KNEE = auto()
+  READ_L_SHOULDER = auto()
+  READ_L_ELBOW = auto()
+  READ_R_SHOULDER = auto()
+  READ_R_ELBOW = auto()
+  READ_CHECKSUM = auto()
 
-from .base_parser import Parser
-from.json_parser import JsonParser
\ No newline at end of file
+
+class JointInformation():
+  """Contains the joint information received from serial connection
+  """
+  def __init__(self):
+    """Initalized with dummy values
+    """
+    self.set_to_default()
+
+  def set_to_default(self):
+    """Sets the object to the default values
+    """
+    self.set_all_joint_values()
+    self.theta_x = 0
+    self.theta_y = 0
+    self.theta_z = 0
+    self.checksum = 0
+    self.checksum_error = False
+    self.packet_available = False
+    self.data_request = False
+    self.at_goal = False
+
+  def __str__(self):
+    return 'left_hip: {}\nleft_knee: {}\nright_hip: {}\nright_knee: {}\n'\
+    'left_shoulder: {}\nleft_elbow: {}\nright_shoulder: {}\nright_elbow: {}\n' \
+    'checksum: {}\nchecksum_error: {}\npacket_available: {}\n' \
+    'data_request: {}'.format(self.left_hip, self.left_knee, self.right_hip,
+    self.right_knee, self.left_shoulder, self.left_elbow, self.right_shoulder,
+    self.right_elbow, self.checksum, self.checksum_error, self.packet_available,
+    self.data_request)
+
+  def set_theta_value(self, s: str, val: int):
+    """Sets the imu theta value
+    """
+    if s == "θx":
+      self.theta_x = val
+    elif s == "θy":
+      self.theta_y = val
+    elif s == "θz":
+      self.theta_z = val
+    else:
+      raise ValueError("Imu axes name should be one of these: θx, θy, θz")
+  
+  def get_theta_value(self, s: str) -> int:
+    """Returns the imu value corresponding to the axes represented by s
+    """
+    if s == "θx":
+      return self.theta_x
+    elif s == "θy":
+      return self.theta_y
+    elif s == "θz":
+      return self.theta_z
+    else:
+      raise ValueError("Imu axes name should be one of these: θx, θy, θz")   
+
+  def set_joint_value_from_state(self, s: SerialReadState, value: int):
+    """Sets the joint value for the given SerialReadState
+    """
+    if s == SerialReadState.READ_L_HIP:
+      self.left_hip = value
+    elif s == SerialReadState.READ_L_KNEE:
+      self.left_knee = value
+    elif s == SerialReadState.READ_R_HIP:
+      self.right_hip = value
+    elif s == SerialReadState.READ_R_KNEE:
+      self.right_knee = value
+    elif s == SerialReadState.READ_L_SHOULDER:
+      self.left_shoulder = value
+    elif s == SerialReadState.READ_L_ELBOW:
+      self.left_elbow = value
+    elif s == SerialReadState.READ_R_SHOULDER:
+      self.right_shoulder = value
+    elif s == SerialReadState.READ_R_ELBOW:
+      self.right_elbow = value
```

Comment:
Noticed that this switch appears a lot in the code. You could make a dictionary like

```python
self.state_map = {
  SerialReadState.Read_L_HIP: self.left_hip,
  SerialReadState.Read_L_KNEE: self.left_knee,
  ...
}
```
Then you can refer to the switch using the `.items()` function. For example, the code above could be reduced to:
 
```suggestion
    for state, joint_value in self.state_map.items():
      if s == state:
        joint_value = value
        return
```